lib/dfe/analytics/rspec/matchers/helpers.rb

```diff
@@ -0,0 +1,29 @@
+# lib/helpers.rb
+
+module DfE
+  module Analytics
+    module RSpec
+      module Matchers
+        # Helpers for matchers in this package.
+        module Helpers
+          def queue_adapter
+            ::ActiveJob::Base.queue_adapter
+          end
+
+          def jobs_to_event_types(jobs)
+            jobs.map do |j|
+              j[:args].first.map do |e|
+                e.fetch('event_type')
+              end
+            rescue StandardError
+              # Parsing the job args above makes a couple of assumptions that may
+              # raise an error. Treat these as non-analytics events by returning
+              # 'nil' for the type.
+              nil
```

Comment:
Should we log here, or are all errors definitely to be ignored?

Reply:
I think they can safely be ignored. I can't see a way to be tripped up by this when using the matcher in your spec. The only way to trip up that comes to mind is that if you're changing the matcher code, you might be a little bit surprised if you have an error in there, but I think that's rare and that if it happens, it won't be hard to debug when you realise that all the `event_types` returned are `nil`.

Do you think that's good enough justification to swallow errors here?